### PR TITLE
Support multi sequencer

### PR DESCRIPTION
**Pull request by gupadhyaya**
fixes https://github.com/rollkit/rollkit/issues/819 and https://github.com/rollkit/rollkit/issues/816

### CODE REVIEW

File: node/full_client_test.go [hunk 1]
```diff
@@ -711,7 +726,7 @@ func TestValidatorSetHandling(t *testing.T) {
 	}
 
 	// 6th EndBlock removes first validator from the list
```
Comment (nashqueue) [inline node/full_client_test.go:728]:
```suggestion
	// 4th EndBlock removes first validator from the list
```

**Review by nashqueue — APPROVED** [review]

File: node/full_client_test.go [hunk 2]
```diff
@@ -721,7 +736,8 @@ func TestValidatorSetHandling(t *testing.T) {
 	}
 
 	// 8th EndBlock adds validator back
```
Comment (nashqueue) [inline node/full_client_test.go:738]:
```suggestion
	// 5th EndBlock adds validator back
```

**Review by nashqueue — APPROVED** [review]

File: block/manager.go [hunk 3]
```diff
@@ -449,13 +462,37 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 	}, nil
 }
 
+func (m *Manager) IsProposer() (bool, error) {
+	// if proposer is not set, assume self proposer
+	if m.lastState.Validators.Proposer == nil {
```
Comment (tuxcanfly) [inline block/manager.go:467]: Consider moving mutex lock / unlock around read here.

**Review by tuxcanfly — APPROVED** [review]
OK.

File: block/manager.go (hunk 3 above)
Comment (gupadhyaya) [reply block/manager.go:467]: it is kind of same as the function itself has multiple reads to `m.lastState` one at the beginning and one at the end. will it make any difference?

**Pull request merged**